snmalloc-sys/build.rs

```diff
@@ -1,335 +1,460 @@
-        } else if triple.contains("i686") {
-            cfg = cfg.define("ANDROID_ABI", "x86");
-        } else if triple.contains("neon") {
-            cfg = cfg.define("ANDROID_ABI", "armeabi-v7a with NEON")
-        } else if triple.contains("arm") {
-            cfg = cfg.define("ANDROID_ABI", "armeabi-v7a");
+        // General Unix handling (excluding FreeBSD and macOS)
+        if config.is_unix() && !cfg!(target_os = "macos") && !cfg!(target_os = "freebsd") {
+            if config.is_gnu() {
+                println!("cargo:rustc-link-lib=c_nonshared");
+            }
+        } else if !config.is_windows() {
+            let cxxlib = if cfg!(any(target_os = "macos", target_os = "openbsd")) {
+                "c++"
+            } else {
+                "stdc++"
+            };
+            println!("cargo:rustc-link-lib={}", cxxlib);
         }
     }
+}
 
-    if cfg!(all(windows, target_env = "msvc")) {
-        cfg = cfg.define("CMAKE_CXX_FLAGS_RELEASE", "/O2 /Ob2 /DNDEBUG /EHsc");
-        cfg = cfg.define("CMAKE_C_FLAGS_RELEASE", "/O2 /Ob2 /DNDEBUG /EHsc");
-        if cfg!(feature = "win8compat") {
-            cfg = cfg.define("WIN8COMPAT", "ON")
+#[cfg(feature = "build_cc")]
+use cc;
+
+#[cfg(feature = "build_cc")]
+fn main() {
+    let mut config = BuildConfig::new();
+    config.builder
+        .include("snmalloc/src")
+        .file("snmalloc/src/snmalloc/override/rust.cc")
+        .cpp(true)
+        .debug(config.debug);
+
+    configure_platform(&mut config);
+    configure_compiler_flags(&mut config);
+    configure_tls(&mut config);
+    configure_features(&mut config);
+
+    config.builder.out_dir(&config.out_dir);
+    config.builder.compile(config.target_lib);
+    configure_linking(&config, None);
+}
+
+#[cfg(feature = "build_cc")]
+fn configure_platform(config: &mut BuildConfig) {
+    if config.is_windows() {
+        config.builder
+            .flag_if_supported("-mcx16")
+            .flag_if_supported("-fno-exceptions")
+            .flag_if_supported("-fno-rtti")
+            .flag_if_supported("-pthread"); // Add pthread support
+
+        if let Some(msystem) = &config.msystem {
+            match msystem.as_str() {
+                "CLANG64" | "CLANGARM64" => {
+                    config.builder
+                        .flag_if_supported("-flto")
+                        .flag_if_supported("-fuse-ld=lld")
+                        .flag_if_supported("-stdlib=libc++")
+                        .flag_if_supported("-Wno-error=unknown-pragmas")
+                        .flag_if_supported("-Qunused-arguments");
+                }
+                "UCRT64" => {
+                    config.builder
+                        .flag_if_supported("-Wno-error=unknown-pragmas")
+                        .flag_if_supported("-fuse-ld=lld")
+                        .flag_if_supported("-Qunused-arguments");
+                }
+                _ => {}
+            }
         }
+    } else if config.is_linux() {
```

Comment:
why just Linux ? e.g. position independent code is a commonality amongst unixes.

Reply:
ok, I have a flag for is unix too so I can add an or, any flags I should be concerned about that should be different between linux/unix

Reply:
unix should be fine, I see you already took care of some edge cases (e.g. haiku and TLS model), so that s good.

Reply:
PIC is now included for both, had some issues with linking libs so those are still different